sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c

```diff
@@ -204,39 +200,26 @@ int gnrc_tcp_init(void)
 
 void gnrc_tcp_tcb_init(gnrc_tcp_tcb_t *tcb)
 {
+    /* Clear TCB with zeros */
+    memset(tcb, 0, sizeof(gnrc_tcp_tcb_t));
 #ifdef MODULE_GNRC_IPV6
     tcb->address_family = AF_INET6;
-    ipv6_addr_set_unspecified((ipv6_addr_t *) tcb->local_addr);
-    ipv6_addr_set_unspecified((ipv6_addr_t *) tcb->peer_addr);
 #else
     tcb->address_family = AF_UNSPEC;
     DEBUG("gnrc_tcp.c : gnrc_tcp_tcb_init() : Address unspec, add netlayer module to makefile\n");
 #endif
     tcb->local_port = PORT_UNSPEC;
     tcb->peer_port = PORT_UNSPEC;
     tcb->state = FSM_STATE_CLOSED;
-    tcb->status = 0;
-    tcb->snd_una = 0;
-    tcb->snd_nxt = 0;
-    tcb->snd_wnd = 0;
-    tcb->snd_wl1 = 0;
-    tcb->snd_wl2 = 0;
-    tcb->rcv_nxt = 0;
-    tcb->rcv_wnd = 0;
-    tcb->iss = 0;
-    tcb->irs = 0;
-    tcb->mss = 0;
-    tcb->rtt_start = 0;
     tcb->rtt_var = RTO_UNINITIALIZED;
     tcb->srtt = RTO_UNINITIALIZED;
     tcb->rto = RTO_UNINITIALIZED;
-    tcb->retries = 0;
-    tcb->pkt_retransmit = NULL;
     tcb->owner = KERNEL_PID_UNDEF;
+    tcb->pkt_retransmit = NULL;
     tcb->rcv_buf_raw = NULL;
+    tcb->next = NULL;
```

Comment:
Absolutely. My comment was more loud thinking that we should fix it at the other places.

However, I always thought that `NULL` equals `0` was enforced by the standard.
Wikipedia states: "The macro NULL is defined as an implementation-defined null pointer constant,[2] which in C99 can be portably expressed as the integer value 0 converted implicitly or explicitly to the type void*." (https://en.wikipedia.org/wiki/Null_pointer#c-std)
 I wasn't aware that ANSI